plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/generic/SQLServerMetaModel.java

```diff
@@ -171,27 +169,52 @@ public String getProcedureDDL(DBRProgressMonitor monitor, GenericProcedure sourc
         }
         if (getServerType() == ServerType.SYBASE) {
             try (JDBCSession session = DBUtils.openMetaSession(monitor, sourceObject, "Read routine definition")) {
-                try (JDBCPreparedStatement dbStat = session.prepareStatement(
-                    "SELECT proc_defn from SYS.SYSPROCEDURE s\n" +
-                        "LEFT JOIN " + DBUtils.getQuotedIdentifier(sourceObject.getCatalog()) + ".dbo.sysobjects so " +
+                if (isMetaDataViewExists(session)) {
+                    try (JDBCPreparedStatement dbStat = session.prepareStatement(
+                        "SELECT source, proc_defn " +
+                        "FROM SYS.SYSPROCEDURE s\n" +
+                        "LEFT JOIN " +
+                        DBUtils.getQuotedIdentifier(sourceObject.getCatalog()) + ".dbo.sysobjects so " +
                         "ON so.id = s.object_id\n" +
                         "WHERE s.proc_name=?"
-                )) {
-                    dbStat.setString(1, objectName);
-                    try (JDBCResultSet dbResult = dbStat.executeQuery()) {
-                        if (dbResult.nextRow()) {
-                            return dbResult.getString(1);
+                    )) {
+                        dbStat.setString(1, objectName);
+                        try (JDBCResultSet dbResult = dbStat.executeQuery()) {
+                            if (dbResult.nextRow()) {
+                                if (dbResult.getString(1) != null) {
+                                    return dbResult.getString(1);
+                                }
+                                return dbResult.getString(2);
+                            }
+                            return "-- Routine '" + objectName + "' definition not found";
                         }
-                        return "-- Routine '" + objectName + "' definition not found";
                     }
                 }
+                return extractSource(monitor,
+                    dataSource,
+                    sourceObject,
+                    sourceObject.getCatalog(),
+                    sourceObject.getSchema().getName(),
+                    objectName);
             } catch (SQLException e) {
                 throw new DBException(e, dataSource);
             }
         }
         return extractSource(monitor, dataSource, sourceObject, sourceObject.getCatalog(), sourceObject.getSchema().getName(), objectName);
     }
 
+    private boolean isMetaDataViewExists(@NotNull JDBCSession session) {
+        boolean result = false;
+        try (Statement dbStat = session.createStatement()) {
+            try (ResultSet resultSet = dbStat.executeQuery("SELECT 1 FROM SYS.SYSPROCEDURE WHERE 1<>1 LIMIT 1")) {
```

Comment:
SQL Server doesn't support LIMIT 🙄 
`SELECT TOP 1 1 FROM SYS.SYSPROCEDURE WHERE 1 <> 1`

Reply:
Ok, I was relying on it
https://www.mysqltutorial.org/mysql-basics/mysql-limit/

Reply:
done